WalletLibrary/WalletLibrary/Errors/TokenValidationError.swift

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+/**
+ * This class is used to represent different types of validation errors related to tokens.
+ * TODO: add a way to inject correlatin vector in all errors.
+ */
+class TokenValidationError: VerifiedIdError
+{
+    /// Creates an instance of `TokenValidationError` when property of the token does not have the expected value.
+    static func InvalidProperty(_ prop: String, actual: String?, expected: String) -> TokenValidationError
+    {
+        if let actual = actual
+        {
+            let message = "Invalid String for property: \(prop). Expected: \(expected), Actual: \(actual)."
+            return TokenValidationError(message: message,
+                                        code: "invalid_property",
+                                        correlationId: nil)
+        }
+        
+        let message = "Propery \(prop) is not present. Expected: \(expected)."
+        return TokenValidationError(message: message,
+                                    code: "invalid_property",
```

Comment:
Would having a separate code for missing property would help?

Reply:
I combined them because it makes it easier to use in the code. For example,

`
        if expectedIssuer != content.iss
        {
            throw TokenValidationError.InvalidProperty("iss", actual: content.iss, expected: expectedIssuer)
        }
`

content.iss could be nil but we only care if it is equal or not so the Error handles the messaging for nil vs a different value instead of us having to check in the code.